packages/hdwallet-walletconnectV2/src/walletconnectV2.ts

```diff
@@ -358,4 +358,22 @@ export class WalletConnectV2HDWallet implements HDWallet, ETHWallet {
   public async getFirmwareVersion(): Promise<string> {
     return "WalletConnectV2";
   }
+
+  public async ethGetChainId(): Promise<number | null> {
+    return this.provider.chainId;
+  }
+
+  public async ethSwitchChain({ chainId }: AddEthereumChainParameter): Promise<void> {
+    const parsedChainId = parseInt(chainId, 16);
+    if (isNaN(parsedChainId) || this.chainId === parsedChainId) {
+      return;
+    }
+
+    this.chainId = parsedChainId;
+    this.provider.chainId = parsedChainId;
```

Comment:
Do we need to mutate this optimistically? `this.provider` being the Ethereum provider, its `chainId` property should automagically reflect the one after the switch?

Reply:
excellent point, actioning now